pallets/marketplace/src/tests/extrinsics.rs

```diff
@@ -599,3 +691,195 @@ fn set_commission_fee_unhappy() {
             assert_noop!(ok, Error::<Test>::NotMarketplaceOwner);
         })
 }
+
+#[test]
+fn update_uri_happy() {
+    ExtBuilder::default()
+        .caps(vec![(ALICE, 10000)])
+        .build()
+        .execute_with(|| {
+            let alice: mock::Origin = RawOrigin::Signed(ALICE).into();
+            assert_eq!(Marketplace::marketplace_id_generator(), 0);
+            assert_eq!(Marketplace::marketplaces(1), None);
+
+            let fee = 25;
+            let name = vec![50];
+            let kind = MPT::Public;
+            let uri = Some(vec![66]);
+            let updated_uri = Some(vec![67]);
+            let info = MarketplaceInformation::new(
+                kind,
+                fee,
+                ALICE,
+                vec![],
+                name.clone(),
+                uri.clone(),
+                uri.clone(),
+            );
+            let updated_info = MarketplaceInformation::new(
+                kind,
+                fee,
+                ALICE,
+                vec![],
+                name.clone(),
+                updated_uri.clone(),
+                uri.clone(),
+            );
+
+            assert_ok!(Marketplace::create(
+                alice.clone(),
+                kind.clone(),
+                fee,
+                name.clone(),
+                uri.clone(),
+                uri.clone()
+            ));
+            assert_eq!(Marketplace::marketplace_id_generator(), 1);
+            assert_eq!(Marketplace::marketplaces(1), Some(info));
+            assert_ok!(Marketplace::update_uri(
```

Comment:
Before this add a one liner that checks that the current uri isn't the same as the future one.
Something like this:
`assert_ne!(Marketplace::marketplaces(1).unwrap().uri, future_uri)`